src/nstime.c

```diff
@@ -141,8 +141,12 @@ nstime_monotonic_t *nstime_monotonic = JEMALLOC_N(n_nstime_monotonic);
 #endif
 bool
 nstime_update(nstime_t *time) {
-	nstime_t old_time;
+	if (nstime_monotonic()) {
+		nstime_get(time);
+		return false;
+	}
 
+	nstime_t old_time;
 	nstime_copy(&old_time, time);
 	nstime_get(time);
 
```

Comment:
@jasone : it looks this function is intended to ensure the *time is updated forward, regardless if clock is monotonic or not. In that case, do you think we should add an assert(!nstime_monotonic()) or fatal in the branch below, and just forget about the change here?